### DISCUSSION TITLE
How to manually rerun a loader without navigating?

### DISCUSSION THREAD

**Discussion by MartinJaskulla** [TanStack/router#204] [Q&A]
Let's say I am on page /posts.

There I can 
- see a list of all my posts
- submit a new post

After submitting a new post, I want to rerun the loader, so that the list of posts gets updated.

How can I do that?

It would be nice to have something like:
```ts
const routeMatch = useMatch<LocationGenerics>();
routeMatch.reload()
```

**Comment by garrett-wombat** [comment]
This feature is really needed. If a pr was created for it would it get merged?

Theres probably a workaround using another query management tool 
- wrap the router context with a data caching context
- trigger data caching context from the loader
- use caching context within the component to get and reload data

This probably bypasses a lot of the loaders functional though (pending states wont be updated, ect)

**Comment by ssalka** [comment]
Also looking for a solution to this. I have a query set up in the loader of a route that is later invalidated by a react-query update mutation. I was originally trying to do `queryClient.invalidateQueries` after the call finishes, however I'm unable to because the query is in fact not in the react-query cache. That's fine, but I'm still wanting some way to invalidate my route's loader, which would cause the query to re-run and then re-render the page.
